http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -201,18 +220,73 @@ protected String getRedirectUri(String state) {
                 return null;
             }
 
-            URIBuilder redirectUriBuilder = new URIBuilder(deployment.getAuthUrl())
-                    .addParameter(RESPONSE_TYPE, CODE)
-                    .addParameter(CLIENT_ID, deployment.getResourceName())
-                    .addParameter(REDIRECT_URI, rewrittenRedirectUri(url))
-                    .addParameter(STATE, state);
-            redirectUriBuilder.addParameters(forwardedQueryParams);
+            String redirectUri = rewrittenRedirectUri(url);
+            URIBuilder redirectUriBuilder = new URIBuilder(deployment.getAuthUrl());
+            redirectUriBuilder.addParameter(RESPONSE_TYPE, CODE)
+                    .addParameter(CLIENT_ID, deployment.getResourceName());
+
+            switch (deployment.getAuthenticationRequestFormat()) {
+                case REQUEST:
+                    if (deployment.getRequestParameterSupported()) {
+                        // add request objects into request parameter
+                        try {
+                            createRequestWithRequestParameter(REQUEST, redirectUriBuilder, redirectUri, state, forwardedQueryParams);
+                        } catch (IOException | JoseException e) {
+                            throw log.unableToCreateRequestWithRequestParameter();
+                        }
+                    } else {
+                        // send request as usual
+                        createOAuthRequest(redirectUriBuilder, redirectUri, state, forwardedQueryParams);
+                        log.requestParameterNotSupported();
+                    }
+                    break;
+                case REQUEST_URI:
+                    if (deployment.getRequestUriParameterSupported()) {
+                        try {
+                            createRequestWithRequestParameter(REQUEST_URI, redirectUriBuilder, redirectUri, state, forwardedQueryParams);
+                        } catch (IOException | JoseException e) {
+                            throw log.unableToCreateRequestUriWithRequestParameter();
+                        }
+                    } else {
+                        // send request as usual
+                        createOAuthRequest(redirectUriBuilder, redirectUri, state, forwardedQueryParams);
```

Comment:
Do you think it would be possible to add a test for this case?

Reply:
I am looking into this. 
This needs to be edited when setting up the client for tests or maybe I can edit the OidcClientConfiguration from the tests to not support request/request-uri.

Reply:
Right, I was thinking about the case where the provider doesn't support it. Is this something that can be turned off with the Keycloak client configuration?